src/channel.c

```diff
@@ -80,6 +78,48 @@ IMPLEMENT_MOCKABLE_FUNCTION(, THANDLE(CHANNEL), channel_create, THANDLE(PTR(LOG_
     return result;
 }
 
+IMPLEMENT_MOCKABLE_FUNCTION(, int, channel_open, THANDLE(CHANNEL), channel)
```

Comment:
>ION(, int, channel_open, THANDLE(CHANNEL), [](http://example.com/codeflow?start=24&length=42)

Why do we need channel_internal again? I thought that it was born BECAUSE we did not do open/close on the channel and <insert long explanation here>

Now that we do have open/close, why do we need this wrapper layer of channel over channel_internal?

Reply:
It can be removed, however it will lead to a big diff.

I wanted to send the smaller change for open/close first so that it is easier to review.

Removal of channel_internal will come in a separate PR later.

#30251622